siliconcompiler/core.py

```diff
@@ -166,20 +167,59 @@ def top(self):
         return entrypoint
 
     ###########################################################################
-    def _get_task(self, step, index='0', flow=None):
+    def _get_tool_task(self, step, index, flow=None):
         '''
-        Helper function to get the name of the task associated with a given step/index.
-        The flowgraph step name may be descriptive for disambiguation pruposes, while the
-        task name defines how the associated tool should be configured and run.
+        Helper function to get the name of the tool and task associated with a given step/index.
+        '''
+        if not flow:
+            flow = self.get('option', 'flow')
 
-        TODO: Is this sort of schema shortcut worth adding?
-              If so should we also add 'get_tool'?
-              Should it be called 'get_step_task'?
+        return self.get('flowgraph', flow, step, index, 'tool'), self.get('flowgraph', flow, step, index, 'task')
+
+    def _get_task(self, step, index, flow=None):
+        '''
+        Helper function to get the name of the task associated with a given step/index.
         '''
+        _, task = self._get_tool_task(step, index, flow=flow)
+        return task
+
+    def _get_tool_module(self, step, index, flow=None, error=True):
+        if not flow:
+            flow = self.get('option', 'flow')
+
+        tool, _ = self._get_tool_task(step, index, flow=flow)
+
+        taskmodule = self.get('flowgraph', flow, step, index, 'taskmodule')
+        module_path = taskmodule.split('.')
+
+        tool_module = '.'.join(module_path[0:-1] + [tool])
+
+        try:
+            return importlib.import_module(tool_module)
```

Comment:
I think this will fully import the module before returning it, which may involve evaluating code and could be slow in some cases. Ditto for this line in `_get_task_module`.